packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -248,3 +221,103 @@ def wait_for_container_response():
         endpoints = dcos_api_session.marathon.get_app_service_endpoints(marathon_config['id'])
         assert len(endpoints) == 1, 'The marathon app should have been deployed exactly once.'
         test_containers(endpoints)
+
+
+def test_metrics_containers_nan(dcos_api_session):
+    """Assert that the metrics API can handle app metric gauges with NaN values."""
+    task_name = 'test-metrics-containers-nan'
+    metric_name = 'test_metrics_containers_nan'
+    marathon_app = {
+        'id': '/' + task_name,
+        'instances': 1,
+        'cpus': 0.1,
+        'mem': 128,
+        'cmd': '\n'.join([
+            'echo "Sending gauge with NaN value to $STATSD_UDP_HOST:$STATSD_UDP_PORT"',
+            'echo "{}:NaN|g" | nc -w 1 -u $STATSD_UDP_HOST $STATSD_UDP_PORT'.format(metric_name),
```

Comment:
You shouldn't need the `-w 1` if you are only sending this once.

Reply:
`-w` sets the connect/read timeout. Is that relevant to the number of times the message is sent?

Reply:
I know I only added it in the script you pulled this from to support calling `nc` multiple times. That said, it doesn't hurt to have it there as a timeout.